src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -4209,34 +4209,43 @@ else if (responseText.contains("a horrible grinning clown head emerges")) {
-        if (responseText.contains("notices the Bowl of Scorpions")) {
-          ResultProcessor.removeItem(ItemPool.BOWL_OF_SCORPIONS);
-          Preferences.increment("_drunkPygmyBanishes");
-        }
-      } else if (monsterName.equals("bugbear robo-surgeon")) {
-        BugbearManager.clearShipZone("Medbay");
-      } else if (monsterName.equals("wumpus")) {
-        WumpusManager.reset();
-      } else if (monsterName.equals("Baron von Ratsworth")) {
-        TavernRequest.addTavernLocation('6');
-      } else if (monsterName.equals("the invader")) {
-        Preferences.setBoolean("spaceInvaderDefeated", true);
-      } else if (monsterName.equals("Eldritch Tentacle")) {
-        Preferences.increment("eldritchTentaclesFought", 1);
-      } else if (special == SpecialMonster.SEWER && !EncounterManager.ignoreSpecialMonsters) {
-        AdventureResult result = AdventureResult.tallyItem("sewer tunnel explorations", false);
-        AdventureResult.addResultToList(KoLConstants.tally, result);
+      // Track monsters defeated on the current PirateRealm island
+      if (adventure == AdventurePool.PIRATEREALM_ISLAND) {
+        Preferences.increment("_pirateRealmIslandMonstersDefeated");
+      }
+
+      switch (monsterName) {
+        case "black pudding" -> Preferences.increment("blackPuddingsDefeated", 1);
+        case "general seal" -> ResultProcessor.removeItem(ItemPool.ABYSSAL_BATTLE_PLANS);
+        case "Frank &quot;Skipper&quot; Dan, the Accordion Lord" -> ResultProcessor.removeItem(
+            ItemPool.SUSPICIOUS_ADDRESS);
+        case "Chef Boy, R&amp;D" -> ResultProcessor.removeItem(ItemPool.CHEF_BOY_BUSINESS_CARD);
+        case "drunk pygmy" -> {
+          if (responseText.contains("notices the Bowl of Scorpions")) {
+            ResultProcessor.removeItem(ItemPool.BOWL_OF_SCORPIONS);
+            Preferences.increment("_drunkPygmyBanishes");
+          }
+        }
+        case "bugbear robo-surgeon" -> BugbearManager.clearShipZone("Medbay");
+        case "wumpus" -> WumpusManager.reset();
+        case "Baron von Ratsworth" -> TavernRequest.addTavernLocation('6');
+        case "the invader" -> Preferences.setBoolean("spaceInvaderDefeated", true);
+        case "Eldritch Tentacle" -> Preferences.increment("eldritchTentaclesFought", 1);
+        case "Glass Jack Hummel" -> Preferences.setBoolean("pirateRealmUnlockedSpyglass", true);
+        case "Red Roger" -> Preferences.setBoolean("pirateRealmUnlockedFlag", true);
+        case "giant giant crab" -> Preferences.setBoolean("pirateRealmUnlockedCrabsicle", true);
+        case "jungle titan" -> Preferences.setBoolean("pirateRealmUnlockedBreastplate", true);
+        case "plastic pirate" -> Preferences.increment(
+            "pirateRealmPlasticPiratesDefeated", 1, 50, false);
+        case "pirate radio" -> Preferences.setBoolean("pirateRealmUnlockedRadioRing", true);
+        default -> {
+          if (IslandManager.isBattlefieldMonster(monsterName)) {
```

Comment:
I would rather have these outside the `default` (i.e. not having a default branch), even though they are all exclusive with the provided monsters, because the developer needing to be aware of that when adding things is not great in my book. For example, the PirateRealm `if` immediately above this switch needs to be outside because of the PirateRealm monsters inside this switch, and such a thing isn't obvious from just reading the code.